### DISCUSSION TITLE
Address Not Recognized on a couple of meetings

### DISCUSSION THREAD

**Discussion by Area18Webservant** [code4recovery/12-step-meeting-list#942] [Q&A] [answered]
I have recently run into an issue in updating meeting addresses.

I go type in the new location and address but whenever I tab out, it reverts back to displaying just the town, state and zip code, and gives me the message "Meetings must have specific address".

It is doing this with only the meetings at 2 separate addresses. All other updates are running smoothly. I have verified the addresses and locations on the web, and am not sure what to do. 

An example of one of the addresses:
New Hope Assembly of God
1030 S Custer St Challis, ID 83226

I have tried inputting every variation of the address I can think of. I also saw some similar postings, but did not see a solution that worked for me. Please help! Our meeting guide is our best tool, and I hate telling the requesters that I am working on the issue and not coming up with anything. 

![Screenshot (8)](https://user-images.githubusercontent.com/114895968/193503715-e11e16fc-99c3-4fcd-a007-e2db39335da7.png)

![Screenshot (9)](https://user-images.githubusercontent.com/114895968/193503731-b76977eb-2191-4659-8b51-18000d73a985.png)

**Comment by joshreisner** [comment, marked as answer]
hi @Area18Webservant - you didn't do anything wrong, google maps just hasn't added this address yet. google search knows about it, and maps seems to know the church, but not the address.

in these cases, [adding an address override](https://wordpress.org/plugins/12-step-meeting-list/#i%20need%20to%20correct%20a%20meeting%20address%20or%20change%20a%20pin%27s%20location) is the best course. 

try adding this code to your theme's functions.php file:

```
if (function_exists('tsml_custom_addresses')) {
    tsml_custom_addresses(array(
        '1030 S Custer St, Challis, ID 83226, USA' => array(
            'formatted_address' => '1030 S Custer St, Challis, ID 83226, USA',
            'city' => 'Challis',
            'latitude' => 44.497533,
            'longitude' => -114.222882,
            'approximate' => 'no',
        ),
    ));
}
```

you might want to confirm the latitude and longitude values. my typical go-to in these cases, https://www.latlong.net/, also seems to be a little hazy on this address.

**Reply by Area18Webservant** [reply]
Thank you so much! Individualized inputs solved for most of the issues, but one remains. I am hoping you might have seen this one as well.

We have a meeting at this Address:

395 W Main Street, Ririe, ID 83443, USA

Whenever I inputed it, it would switch to 395 W Main Street Rigby, ID 83442. While having similar names, Ririe and Rigby are 12 miles apart. Google recognizes them as separate locations, but not our meeting guide.

I inputed that that address with the correct Lat/Lon in our theme editor, and when I went to input it in the listing, it finally seemed to recognize it as I tabbed out, But as soon as I hit update, it reverted back to the Rigby Address. I am miffed. 

Thanks for the help given, and any additional help you can provide with this final issue!

**Reply by joshreisner** [reply]
What's the place that the meeting is at? I'm looking at a map of Ririe and I can't seem to find any buildings on West Main St.

**Reply by Area18Webservant** [reply]
It's called the Ririe Senior Citizens Center. I transposed addresses in my original reply. The address is 395 Main Street, Ririe, ID, 83443. Google recognizes the address and location, but seemingly not the Meeting Guide.
![2022-10-04 (2)](https://user-images.githubusercontent.com/114895968/193945154-5eda3234-14ba-45a0-a6d7-e54c1451b50c.png)

**Reply by joshreisner** [reply]
Ok! I think the code you need is 

```
if (function_exists('tsml_custom_addresses')) {
    tsml_custom_addresses(array(
        '395 Main St, Ririe, ID 83443, USA' => array(
            'formatted_address' => '395 Main St, Ririe, ID 83443, USA',
            'city' => 'Ririe',
            'latitude' => 43.6304513,
            'longitude' => -111.7737957,
            'approximate' => 'no',
        ),
    ));
}
```

Then fix the meeting by pasting `395 Main St, Ririe, ID 83443, USA` into the address field exactly, and it should work 🤞 

For what it's worth, on this one the geocoding response is coming back with two results, the first one is Rigby the second one is Ririe.

**Reply by Area18Webservant** [reply]
That's what I saw as well, I just didn't know how to fix it. Your solution worked a charm. Thank you so much for all your help.
